auslib/web/admin/emergency_shutoff.py

```diff
@@ -0,0 +1,104 @@
+import json
+import time
+from flask import Response
+from connexion import problem
+from auslib.global_state import dbo
+from auslib.web.admin.views.base import (
+    requirelogin, handleGeneralExceptions, transactionHandler)
+from auslib.web.admin.views.scheduled_changes import (
+    EnactScheduledChangeView, ScheduledChangesView, SignoffsView)
+
+
+def get_emergency_shutoff(product, channel):
+    where = dict(product=product, channel=channel)
+    shutoffs = dbo.emergencyShutoffs.select(where=where)
+    return shutoffs[0] if shutoffs else None
+
+
+def shutoff_already_exists(product, channel):
+    return bool(get_emergency_shutoff(product, channel))
+
+
+def product_requires_signoffs(product, channel):
+    rs = dbo.productRequiredSignoffs.select(where=dict(product=product, channel=channel))
+    return bool(rs)
+
+
+def process_shutoff_post(emergency_shutoff):
+    if 'csrf_token' in emergency_shutoff:
+        del emergency_shutoff['csrf_token']
+    if 'data_version' not in emergency_shutoff:
+        emergency_shutoff['data_version'] = 1
+    return emergency_shutoff
+
+
+@requirelogin
+@transactionHandler
+@handleGeneralExceptions('POST')
+def post(emergency_shutoff, changed_by, transaction):
+    if shutoff_already_exists(emergency_shutoff['product'], emergency_shutoff['channel']):
+        return problem(
+            400, 'Bad Request', 'Invalid Emergency shutoff data',
+            ext={'data': 'Emergency shutoff for product/channel already exists.'})
+    if not product_requires_signoffs(emergency_shutoff['product'], emergency_shutoff['channel']):
+        return problem(
+            400, 'Bad Request', 'Invalid Emergency shutoff data',
+            ext={'data': 'The given product/channel should requires signoffs.'})
+    emergency_shutoff = process_shutoff_post(emergency_shutoff)
```

Comment:
I think it would be better to get rid of this and explicitly pass product and channel. I don't think there's a reason to be setting data_version here (correct me if I'm wrong though - maybe deleting & recreating rows with the same PK works different), and past that the helper provides little value.